recipes-core/acrn/acrn-hypervisor.bb

```diff
@@ -2,10 +2,12 @@ require acrn-common.inc
 
 ACRN_BOARD ?= "nuc7i7dnb"
 ACRN_SCENARIO  ?= "industry"
+ACRN_BOARD_FILE ?= ""
+ACRN_SCENARIO_FILE ?= ""
 
 EXTRA_OEMAKE += "HV_OBJDIR=${B}/hypervisor "
-EXTRA_OEMAKE += "BOARD=${ACRN_BOARD} SCENARIO=${ACRN_SCENARIO}"
-EXTRA_OEMAKE += "BOARD_FILE=${S}/misc/acrn-config/xmls/board-xmls/${ACRN_BOARD}.xml SCENARIO_FILE=${S}/misc/acrn-config/xmls/config-xmls/${ACRN_BOARD}/${ACRN_SCENARIO}.xml"
+EXTRA_OEMAKE += "${@bb.utils.contains('ACRN_BOARD_FILE', '', 'BOARD_FILE=${ACRN_BOARD_FILE}', 'BOARD=${ACRN_BOARD}', d)}"
```

Comment:
If xmls not present, then it should be build using source method !, then condition should be like below ?

EXTRA_OEMAKE += "${@bb.utils.contains('ACRN_BOARD_FILE', '', 'BOARD=${ACRN_BOARD}' ,'BOARD_FILE=${ACRN_BOARD_FILE}' , d)}"
EXTRA_OEMAKE += "${@bb.utils.contains('ACRN_SCENARIO_FILE', '','SCENARIO=${ACRN_SCENARIO}', 'SCENARIO_FILE=${ACRN_SCENARIO_FILE}', d)} "

Reply:
contains() check against empty/blank variable will always return false. 
in this case ACRN_BOARD_FILE  is empty by default so contains() is false which is "BOARD=${ACRN_BOARD}"